### DISCUSSION TITLE
[Chapter 03] 로케일 클래스에서의 KOREA vs. KOREAN

### DISCUSSION THREAD

**Discussion by corock** [cowaine/is-my-code-that-strange#13] [Chapter 01-03 (설계)] [answered]
코드 3.8(_p. 30_)에서는 `Currency` 클래스의 인스턴스를 가져올 때 로케일(_locale_)을 `KOREA` 로 설정하고 있습니다.

```java
Currency won = Currency.getInstance(Locale.KOREA);
Money money = new Money(100, won);
money.amount = -200;
```

`Locale` 클래스 내부를 살펴보면 `KOREA` 외에도 `KOREAN` 상수도 있습니다.

```java
public final class Locale implements Cloneable, Serializable {

    // (...)

    public static final Locale KOREAN;

    // (...)

    public static final Locale KOREA;

    // (...)

}
```

두 상숫값 중 아무거나 사용해도 되는 걸까요? 만약 아니라면 각각 어떤 상황에서 사용하면 좋을지 궁금합니다.

**Comment by firewoody237** [comment, marked as answer]
https://docs.oracle.com/javase/8/docs/api/java/util/Locale.html

여기보면, `field described`를 language와 contry를 따로 정의하고 있고,

생성자에서도 language와 country를 (String 타입이지만) 따로 받고 있습니다.

그리고 메소드에서도 `getCountry()`나 `getLanguage()`처럼 따로 정의하고 있기 때문에, 
작성하신 코드를 사용하는 곳에서 문제가 발생하지 않도록 사용하는 Locale이 "언어"인지 "국가"인지 확실하게 사용하는게 좋을 것 같아 보입니다!

**Comment by corock** [comment]
_cf._ [Locale 관리](https://goodteacher.tistory.com/723)
